django-rgd/rgd/filters.py

```diff
@@ -74,6 +76,14 @@ class SpatialEntryFilter(filters.FilterSet):
         method='filter_time_of_day',
     )
 
+    collections = filters.ModelMultipleChoiceFilter(
+        help_text='One or more collections that the data might belong to.',
+        label='Collections',
+        method='filter_collections',
+        queryset=Collection.objects.all(),
+        field_name='files__collection',
```

Comment:
Unfortunately, this is not sufficient. Downstream models do not all implement a `files` field and so we have to go through the utilities in `permissions.py` to get the paths to the file resources for each downstream model. This is rather complicated and turned out to be quite a bit of work to get right; which is on me for not foreseeing when assigning this to you.